### DISCUSSION TITLE
Convert: create labels with names below thumbnails like those in "montage"

### DISCUSSION THREAD

**Discussion by Forgev41** [ImageMagick/ImageMagick#1906] [General]
My goal is to create a composite image with several people on a background.
Below the thumbnails I would like to have the names of the people.

For several hours I've been wrestling with label and annotate, I don't get it right.
All the examples on the forum and on the internet use one image, but I use between 10 and 20.

I know that I can use montage and this works fine, however I would like to position the thumbnails myself.

Question: how do I generate labels below the images with magick convert like the ones created with magick montage

This is the montage image with the labels I would like to have in convert:
https://i.stack.imgur.com/W8HXr.jpg
`magick montage -font Comic-Sans-MS-Bold -label "%t" *.jpg -border 5 -shadow -geometry 386x224+10+10 -background white contacts.png`

This is the image I created with magick convert with several small thumbnails on a background.
So I would like to add a label below the thumbnails.
https://i.stack.imgur.com/Anpnq.jpg
`magick convert "Rodion Kutsaev.png" -resize 2657x1795 ^
( "Ana Maria.jpg" -auto-orient -resize "299x448^" ) -geometry +210+169 -composite ^
( "Brooke-Cagle 1.jpg" -auto-orient -resize "299x448^" ) -geometry +529+169 -composite ^
( "Brooke Cagle.jpg" -auto-orient -resize "299x448^" ) -geometry +849+169 -composite ^
( "Gabriel Silverio.jpg" -auto-orient -resize "299x448^" ) -geometry +1165+169 -composite ^
( "Reinaldo Kevin.jpg" -auto-orient -resize "299x448^" ) -geometry +1486+169 -composite ^
-gravity North -font Comic-Sans-MS-Bold -pointsize 48 ^
-fill gray -annotate +1+66 "My titlel" ^
-fill gray -annotate +2+67 "My titlel" ^
-fill gray -annotate +3+68 "My titlel" ^
-fill white -annotate +0+65 "My titlel" ^
group_v2.png`

 BTW:
1) since I'm on Windows I use the ^ instead of the \
2) all images where take from the amazing unsplash.com website.

**Comment by fmw42** [comment]
On way to resize and label each image is to do montage on one image at a time.

`magick montage lena.jpg -geometry 128x128+0+0 -set label "testing" -tile 1x1 x.png`

![x](https://user-images.githubusercontent.com/13458218/79644398-9d58b700-815d-11ea-8b97-709545b70cb1.png)

The other way, is to use label: with parenthesis aside processing to generate a text image and append it below the resized image.

magick lena.jpg -resize 128x128 -set option:wd "%w" \( -size "%[wd]x15" -background white -fill black -gravity center label:"testing" \) -append x.png


![x](https://user-images.githubusercontent.com/13458218/79644794-06d9c500-8160-11ea-9616-5d3b9f5cc349.png)

P.S. When using Imagemagick 7, use magick, not magick convert. You still have to preface the other tools with magick, such as magick montage.
